### PR TITLE
Add editTopo.py tool

### PR DESCRIPTION
**Pull request by micaeljtoliveira**

### CODE REVIEW

**Comment by aekiss** [comment]
Is this the latest version from here? https://github.com/COSIMA/topogtools/blob/6b4983127aa18dfdf1b62b2e18b581d82d4a64d4/editTopo.py

**Comment by micaeljtoliveira** [comment]
Yes, I think so.

File: editTopo.py [hunk 1]
```diff
@@ -0,0 +1,908 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+#!/usr/bin/env python
```
Comment (aekiss) [inline editTopo.py]: `#!/usr/bin/env python`
this needs to be the first line in order to run from the CLI (I think!)

Comment (micaeljtoliveira) [reply editTopo.py]: Good catch! Fixed.

**Comment by aekiss** [comment]
If so, it might be good to include the commit hash or equivalently the URL https://github.com/COSIMA/topogtools/blob/6b4983127aa18dfdf1b62b2e18b581d82d4a64d4/editTopo.py in the git log to help with version control, since this script is now in 3 places and I'm not sure which of them should be thought of as definitive if we make improvements.

**Comment by micaeljtoliveira** [comment]
@aekiss Good point! I've just added the url to the commit log.

I would suggest we treat the different versions in the following way:
- the one here: used for OM3 and related models/configs.
- the one in topogtools: used for OM2 and related models/configs

The same goes for all the other tools I think. Does this sound reasonable?

**Comment by aekiss** [comment]
Yes that makes sense. It will be tricky to avoid divergence between the versions if we make changes, but it would be nice to keep them synchronised.

**Comment by micaeljtoliveira** [comment]
My thinking was that we wouldn't need to update the tools used for OM2 very often, if at all, because AFAIK, there are no planned updates to the OM2 topography.

**Comment by aekiss** [comment]
Fair point. So the version here should be though of as definitive/latest.

**Comment by aekiss** [comment]
We could put a note to that effect in the topogtools README

**Comment by micaeljtoliveira** [comment]
> We could put a note to that effect in the topogtools README

Yes, I think that's a good idea.

I now realize I should have discussed this in more detail with you at some point, so that we would be on the same page. Sorry about that.

**Comment by aekiss** [comment]
no problem!

**Comment by micaeljtoliveira** [comment]
@aekiss I've now created an [issue](https://github.com/COSIMA/topogtools/issues/11) about the topogtools README.

Anything else needed here before merging this?

File: README.md [hunk 2]
```diff
@@ -189,7 +189,15 @@ usage: png2nc.py
...
-# Building and Installation
+## editTopo.py
+
+`editTopo.py` provides a GUI for hand-editing `topog.nc` files, recording every
+change, and applying these changes to other files. This is an updated version of
+Alistair Adcroft's `editTopo.py` from
+[here](https://github.com/aekiss/MOM6-examples/blob/1c3dc5216139f84b20ce3a5d8ea758bdc7912e8e/ice_ocean_SIS2/OM4_025/preprocessing/editTopo.py)
+and is under a [LGPLv3
+license](https://github.com/NOAA-GFDL/MOM6-examples/blob/dev/gfdl/LICENSE.md).
+For usage details, do `./editTopo.py -h`.  # Building and Installation
```
Comment (aekiss) [inline README.md]:
```suggestion
For usage details, do `./editTopo.py -h`. 
```

File: README.md [hunk 3]
```diff
@@ -189,7 +189,15 @@ usage: png2nc.py
 Converts `test_topo.png` to `test_topo.nc` for use as a test input file for `topogtools deseas`.
 
 
-# Building and Installation
+## editTopo.py
+
+`editTopo.py` provides a GUI for hand-editing `topog.nc` files, recording every
+change, and applying these changes to other files. This is an updated version of
+Alistair Adcroft's `editTopo.py` from
+[here](https://github.com/aekiss/MOM6-examples/blob/1c3dc5216139f84b20ce3a5d8ea758bdc7912e8e/ice_ocean_SIS2/OM4_025/preprocessing/editTopo.py)
```
Comment (aekiss) [inline README.md]:
```suggestion
change, and applying these changes to other files. This was initially copied from 
[here](https://github.com/COSIMA/topogtools/blob/6b4983127aa18dfdf1b62b2e18b581d82d4a64d4/editTopo.py),
which is in turn an updated version of Alistair Adcroft's `editTopo.py` from
[here](https://github.com/aekiss/MOM6-examples/blob/1c3dc5216139f84b20ce3a5d8ea758bdc7912e8e/ice_ocean_SIS2/OM4_025/preprocessing/editTopo.py)
```

**Comment by aekiss** [comment]
A couple more suggestions, but other than that it's ready to merge if the only thing you changed in the code was the GPL notice at the start (ie so I don't need to check the code).

**Comment by micaeljtoliveira** [comment]
Yes, I've only added the copyright notice, everything else should be unchanged. Thanks for the review!

**Review by aekiss — APPROVED** [review]

**Pull request merged**